### DISCUSSION TITLE
QEMU builds for other architectures

### DISCUSSION THREAD

**Discussion by sajattack** [pspdev/pspdev#56] [Ideas]
2 years ago I set up some builds for non-x86_64 architectures on my fork 
https://github.com/sajattack/pspdev/blob/master/.github/workflows/foreign-arch.yml

Unfortunately building the entire toolchain in qemu takes 3+ hours (per arch).

How should we pursue this going forward?

Should we wait for github to add native runners for all these arches or take a different approach?

Is there any benefit to what I set up that we can take away?

**Comment by sharkwouter** [comment]
So it takes 3 times as long? Yeah, we might not want to build this every time. We could only build for releases, but that would mean we're not testing it. Kind of a difficult problem.

**Comment by tpimh** [comment]
Is it possible to run GitHub actions on servers other than GitHub's? If so, I can probably provide the hardware that can do it natively (or in QEMU in less than 3 hours).

**Reply by sharkwouter** [reply]
Yes, it is possible to run on an own runner

**Comment by diamant3** [comment]
There's [linux/arm64](https://github.blog/changelog/2025-01-16-linux-arm64-hosted-runners-now-available-for-free-in-public-repositories-public-preview/) runner now offered by GitHub and hoping we could add it to our build.
